### PR TITLE
Chore/add storybook

### PR DESCRIPTION
**Pull request by JaneMoroz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,28 +11,47 @@
   "scripts": {
     "dev": "vite",
     "build": "vite build && tsc && node copy-types.js",
+    "build:css": "tailwindcss -m -i ./src/tailwind-entry.css -o ./dist/styles.css",
+    "preview": "vite preview",
+    "storybook": "concurrently \"pnpm storybook:css\" \"storybook dev -p 6006\"",
+    "storybook:css": "tailwindcss -w -i ./src/tailwind-entry.css -o ./src/index.css",
+    "build-storybook": "concurrently \"pnpm build-storybook:css\" \"storybook build\"",
+    "build-storybook:css": "tailwindcss -m -i ./src/tailwind-entry.css -o ./src/index.css",
```
Comment (Dan-Y-Ko) [inline package.json]: tailwind-entry.css isn't necessary. It's the same file as the style.css. Remove the tailwind-entry.css file and replace the scripts here with style.css

File: package.json [hunk 2]
```diff
@@ -11,28 +11,47 @@
...
     "@types/react": "^18.3.12",
     "@vitejs/plugin-react": "^4.3.3",
     "autoprefixer": "^10.4.20",
+    "concurrently": "^9.1.0",
     "husky": "^9.1.7",
     "postcss": "^8.4.49",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "semantic-release": "^24.2.0",
+    "storybook": "8.4.6",
```
Comment (Dan-Y-Ko) [inline package.json]: Im getting an error about an unmet peerdepedency with storybook. Seems to require 8.4.7

File: tsconfig.json [hunk 3]
```diff
@@ -3,17 +3,13 @@
     "target": "ESNext",
     "module": "ESNext",
     "jsx": "react-jsx",
-    "declaration": true,
-    "declarationDir": "dist/types",
+    "emitDeclarationOnly": false,
```
Comment (Dan-Y-Ko) [inline tsconfig.json]: Please revert the tsconfig back to the original, it is needed

**Review by Dan-Y-Ko — CHANGES_REQUESTED** [review]
Let's keep the folder naming consistent and use kebab-casing (for multiple words, like avatar-group)

I'm guessing the index.css file is just for storybook? Might be more appropriate to name it something more specific if so

For the ts errors was it something like this

```
The inferred type of 'Outline' cannot be named without a reference to '.pnpm/@storybook+csf@0.1.12/node_modules/@storybook/csf'. This is likely not portable. A type annotation is necessary.
```

I was able to resolve those by just adding the types that were already in the file. So like add Story type to the stories and add the meta type to the meta object.

I'm currently having an issue with @heroicons/react package when I try to install this package in a test project, I'm not sure why. I am trying to figure that out atm

**Comment by Dan-Y-Ko** [comment]
nvm, I got the issue corrected. Everything is working properly on my side with the fixes I mentioned implemented, so that's all that needs to be taken care of

**Comment by Dan-Y-Ko** [comment]
ok so I tested in a nextjs project and it has an issue. The width and height props on the avatar component don't really work correctly. We have to manually pass the that props to the Image component from nextjs, so we need another solution. Maybe get rid of the width and height prop from the avatar component and have it just handled from the consumer directly?

Nope, that won't work. We need the div with the bg-base-200 to handle transparent images but that causes it to stretch the entire screen. I think there's a way to make the nextjs image take the dimensions of the parent div

**Comment by JaneMoroz** [comment]
> make the nextjs image take the dimensions of the parent div

So, I removed width/height from the 'Avatar', I think we have to set Nextjs image's heigth and width everytime we use it, without width/heigth, it throws the error. I tested it in 'AuthHeader' and 'VoteCard' in 'ideation' and haven't encountered any `stretching`

**Comment by Dan-Y-Ko** [comment]
> > make the nextjs image take the dimensions of the parent div
> 
> So, I removed width/height from the 'Avatar', I think we have to set Nextjs image's heigth and width everytime we use it, without width/heigth, it throws the error. I tested it in 'AuthHeader' and 'VoteCard' in 'ideation' and haven't encountered any `stretching`

I'm guessing the reason you don't see it is because you have it inside a div in the component that is constraining it?

Here's a screenshot where it's just by itself. Look carefully, it's hard to see but you there is a different shade extending the entire width.

screenshot

Do you think it's ok to leave it to the outside component to set the boundaries? It may cause weird behaviors

For example, using this code

```
<div className="w-32">
      <Avatar>
        <Image
          alt="avatar"
          src="https://gravatar.com/avatar/a6416cf1e8d0208251a732a6af75530878cdfd92b85d2de9ba6c4fec92d8a157?s=200&r=g&d=robohash"
          width={60}
          height={60}
        />
      </Avatar>
    </div>
```

it looks like this

screenshot

What I was thinking was if there's a way to adapt the size of the image to the surrounding element, that would be ideal (and not set the width/height directly on the nextjs image)

**Comment by Dan-Y-Ko** [comment]
I tried figuring this out but css is just not my strength lol

If you can come up with a different solution to handle transparent images, we can go with that. That's what the bg-base-200 is for.

I guess we could just create an image component in the user dashboard project with the nextjs image component and add the bg-base-200 in there? and remove it from the avatar component styling

**Comment by JaneMoroz** [comment]
@Dan-Y-Ko added `h-fit w-fit` so now the avatar's heigth/width will adjust to the inner content height/width (`<Image/>` in our case)

And about transparent images.... we do have `bg-base-200` added to `Avatar` and you can even add a border with `withBorder` prop, no need to add `bg-base-200` anywhere else... so not sure what you meant here:

> If you can come up with a different solution to handle transparent images, we can go with that. That's what the bg-base-200 is for.

>I guess we could just create an image component in the user dashboard project with the nextjs image component and add the bg-base-200 in there? and remove it from the avatar component styling

**Review by Dan-Y-Ko — APPROVED** [review]
gj :)

**Pull request merged**